### ISSUE TITLE
Refactor the EventingBackend CRD

### ISSUE THREAD

**Issue by nachtmaar** [kyma-project/kyma#14305]
**Description**

While reviewing the Proposal [reimplement backend-crd.md #667](https://github.com/kyma-project/community/pull/667) I discovered a few things on the `EventingBackend` CR which can be improved IMHO.

## Tasks:
As part of this issue we should:
- [ ] Agree if we want to perform any of the proposed changes to the `EventingBackend` CR
- [ ] Implement the changes or create follow-up tickets (if the change is too big)

### Rename the eventingbackend.status.eventingReady field

Before: `status.eventingReady`
After: `status.ready`

> **NOTE:** This was discussed [here](https://github.com/kyma-project/community/pull/667#discussion_r871692732). The team mostly agreed to the renaming, but it was out of scope for the proposal (because the proposal did not touch existing fields of the EventingBackend CR).

<details>
  <summary>Reasoning for the renaming</summary>

The status of the EventingBackend looks like this:
```yaml
status:
  backendType: nats
  conditions:
  - lastTransitionTime: "2022-05-06T08:35:11Z"
    reason: Backend ready
    status: "True"
    type: Backend Ready
  - lastTransitionTime: "2022-05-06T08:35:11Z"
    reason: Publisher proxy deployment ready
    status: "True"
    type: Publisher Proxy Ready
  - lastTransitionTime: "2022-05-06T08:35:11Z"
    reason: Subscription controller started
    status: "True"
    type: Subscription Controller Ready
  eventingReady: true
```

I feel that the name `eventingReady` is not ideal. `eventingReady` is true if all conditions are true (just some syntactic sugar on the API). I propose to rename it from `eventingReady` to `ready`. The name EventingBackend also mentions that it is about eventing. There is no reason (for me) to mention the extra prefix `eventing`. 

</details> 

### Change the EventingBackend label

Before: `kyma-project.io/eventing: backend`
After: `eventing.kyma-project.io/backend: <backend>` where backend=(nats|bab)

> **NOTE:** This was discussed [here](https://github.com/kyma-project/community/pull/667#discussion_r871692923). The team agreed to the renaming, but it was out of scope for the proposal (because the proposal did not touch existing fields of the EventingBackend CR).

<details>
  <summary>Reasoning for the renaming</summary>

The status of the EventingBackend looks like this:
```yaml
apiVersion: eventing.kyma-project.io/v1alpha1
kind: EventingBackend
metadata:
  labels:
    kyma-project.io/eventing: backend
```

We [recently](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/deployment/publisher_deployment.go#L31) introduced the label `eventing.kyma-project.io/backend: <backend>` where backend=(nats|bab). The label is quite handy because it can already be used to identify the correct EPP instance (for the given <backend>). We can bring this label to the `EventingBackend` as well. 
The vision would be that we can query all eventing components for a given <backend> at some point: eventing-publisher-proxy deployment, eventing backend controller, eventing secrets, etc.
Atm the label is pretty useless because every EventingBackend CR has the same label value.
</details>

**Comment by k15r** [comment]
The Backend CRD will be replaces by the new Eventing CRD (from the modularization topic)

**Comment by k15r** [comment]
replaced by eventing cr (modularization)
